iterations_runners/iterations_runner.java

```diff
@@ -5,6 +5,97 @@
 // For this reason we use JNI to make a call to clock_gettime() ourselves.
 
 import java.util.Arrays;
+import java.util.List;
+
+// Instrumentation imports
+import java.lang.management.ManagementFactory;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import java.lang.management.ManagementFactory;
+import java.lang.management.CompilationMXBean;
+import java.lang.management.GarbageCollectorMXBean;
+
+
+class KrunJDKInstrumentation {
+    private CompilationMXBean compBean;
+
+    public KrunJDKInstrumentation() {
+        this.compBean = ManagementFactory.getCompilationMXBean();
+    }
+
+    // Emit cumulative compilation time in milliseconds to stderr
+    private String getCompJSON() {
+        return "" + this.compBean.getTotalCompilationTime();
+    }
+
+
+    // Get GC info for the specified collector
+    private String getGCJSON(GarbageCollectorMXBean gcBean) {
+        String json = "[";
+
+        // Memory pools
+        json += "[";
+        String[] poolNames = gcBean.getMemoryPoolNames();
+        for (int i = 0; i < poolNames.length; i++) {
+            json += '"' + poolNames[i] + '"';
+            if (i < poolNames.length - 1) {
+                json += ", ";
+            }
+        }
+        json += "], ";
+
+        // Collection time and counts
+        json +=  gcBean.getCollectionTime() + ", ";
+        json +=  gcBean.getCollectionCount();
+
+        json += "]";
+        return json;
+
+    }
+
+    // Get GC info for each collector
+    private String getGCJSON() {
+        // Ask for a new list of collectors each time, in-case they change.
+        List<GarbageCollectorMXBean> gcBeans =
+            ManagementFactory.getGarbageCollectorMXBeans();
+
+        String json = "{";
+        for (int i = 0; i < gcBeans.size(); i++) {
+            GarbageCollectorMXBean gcBean = gcBeans.get(i);
+            json += "\"" + gcBean.getName() + "\": " + this.getGCJSON(gcBean);
+            if (i < gcBeans.size() - 1) {
+                json += ", ";
+            }
+        }
+        return json += "}";
+    }
+
+    public String getInstJSON(int iterNum) {
+        /* Build a JSON compatible string which Krun can parse in later.
+         *
+         * The JSON is of the form:
+         * [iterNum, cumuCompTime, [collector1Info, ..., collectorNInfo]]
+         *
+         * Each collectorInfo record is of the form:
+         * [collectorName: [[PoolName1, ..., PoolNameN],
+         *                  cumuCollectTime, cumuCollectCount]
+         *
+         * Here "cumu" means "cumulative" and times are in milliseconds.
+         *
+         * Each collector manages one or more memory pools, and one memory
+         * pool may be managed by multiple collectors.
+         *
+         */
+
```

Comment:
No need for a blank line here, I think. I'm not really sure why the comment is within the block and not a Javadoc above the method definition?